src/main/java/org/osc/manager/ism/api/IsmAgentApi.java

```diff
@@ -16,77 +16,65 @@
+            EntityManager em) throws Exception {
+        this.txControl = txControl;
+        this.em = em;
+        this.vs = vs;
+        this.api = new IsmDeviceApi(this.vs, this.txControl, this.em);
     }
 
-    public static IsmAgentApi create(ApplianceManagerConnectorElement mc, VirtualSystemElement vs) throws Exception {
-        return new IsmAgentApi(mc, vs);
+    public static IsmAgentApi create(ApplianceManagerConnectorElement mc, VirtualSystemElement vs,
+            TransactionControl txControl, EntityManager em) throws Exception {
+        return new IsmAgentApi(mc, vs, txControl, em);
     }
 
     @Override
-    public List<ManagerDeviceMemberStatusElement> getFullStatus(List<DistributedApplianceInstanceElement> list) {
-        return getAgentFullStatus(list);
-    }
-
-    private List<ManagerDeviceMemberStatusElement> getAgentFullStatus(List<DistributedApplianceInstanceElement> list) {
-        // Get basic status
-        ManagerDeviceMemberStatusElement agentElem = null;
+    public List<ManagerDeviceMemberStatusElement> getFullStatus(List<DistributedApplianceInstanceElement> daiList) {
         List<ManagerDeviceMemberStatusElement> response = new ArrayList<>();
-        if (list != null){
-            for (DistributedApplianceInstanceElement dai: list){
-                agentElem = getAgentBasicStatus(dai);
-                response.add(agentElem);
-            }
+        if (daiList == null) {
+            throw new IllegalArgumentException("The provided DAIs should not be null");
         }
-        return response;
-    }
-
-    private ManagerDeviceMemberStatusElement getAgentBasicStatus(DistributedApplianceInstanceElement dai) {
-        IsmAgent agentElem = new IsmAgent();
 
-        SimpleDateFormat format = new SimpleDateFormat("yyyy/MM/dd-hh:mm");
-        String buildTime = format.format(new Date());
-        String ver = String.format("1.2(Build %d, %s)", 1234, buildTime);
+        for (DistributedApplianceInstanceElement dai : daiList) {
+            DeviceMemberEntity member = null;
 
-        agentElem.setDistributedApplianceInstanceElement(dai);
-        agentElem.setVersion(ver);
-        agentElem.setApplianceIp("1.1.1.1");
-        agentElem.setApplianceName("ApplianceName");
-        agentElem.setManagerIp("2.2.2.2");
-        agentElem.setBrokerIp("3.3.3.3");
-        agentElem.setApplianceGateway("1.1.1.1");
-        agentElem.setApplianceSubnetMask("255.255.255.0");
-        agentElem.setRx(ThreadLocalRandom.current().nextLong(10000, 100000));
-        agentElem.setDropSva(ThreadLocalRandom.current().nextLong(0, 1000));
-        agentElem.setTxSva(ThreadLocalRandom.current().nextLong(10000, 100000));
+            try {
+                member = (DeviceMemberEntity) this.api.findDeviceMemberByName(dai.getName());
+            } catch (Exception e) {
+                LOG.error(String.format("Finding device member name %s", dai.getName()), e);
```

Comment:
we have to define the behaviour better. Should we throw an exception and not process further statuses or add a error status to the list?

I dont like returning null. @emanoelxavier your thoughts? We have to use the same behavior for all of our API

Reply:
Some options for this case are:
1.  Add an *empty* device member status element, with all the fields null/empty except the name. This indicates no status found for that device name.
2. Do not add a device at all but return what was found. This is too silent IMO.
3. Throw an exception. Too aggressive IMO since some of the data is available. 

I recommend option 1.

Reply:
OK, implemented option 1.